src/test/java/org/graylog/plugins/usagestatistics/TestClusterConfigService.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2015 Graylog, Inc. (hello@graylog.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.graylog.plugins.usagestatistics;
+
+import com.google.common.collect.Maps;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.graylog2.shared.utilities.AutoValueUtils;
+
+import java.util.Map;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+public class TestClusterConfigService implements ClusterConfigService {
+    private final Map<String, Object> data = Maps.newConcurrentMap();
```

Comment:
It's in most cases an unnecessary indirection (and thus might be slower in performance critical code paths; not the case here, though). Those factory methods were really useful in Java 6 but since the diamond operator (`<>`) has been introduced in Java 7 the language syntax isn't as painful as before. There are cases in which those static factory methods are useful (e. g. initializing a collection with `Iterable<?>`) but most of the time they're not useful anymore.

Also see https://github.com/google/guava/wiki/CollectionUtilitiesExplained#static-constructors

> If we question the usage of these helpers, we should do a cleanup in all of our repositories.

Yes.